File: DoohAdManager.Model/Application/INV/Screen.cs
using DoohAdManager.Model.Shared;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DoohAdManager.Model.Application.INV
{
    public class MvScreen
    {
        public int Id { get; set; }
        public int TenantId { get; set; }
        public required string Name { get; set; }
        public required string Address { get; set; }
        public required string Location { get; set; }
        public required string Resolution { get; set; } 
        public ScreenStatus Status {  get; set; }
        public ScreenOrientation Orientation { get; set; }
        public List<string>? Tag { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public int? CreatedBy { get; set; }
        public int? UpdatedBy { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedAt { get; set; }
        public int? DeletedBy { get; set; }
        public List<MvScreenOperatingHour>? OperatingHours { get; set; }

    }

    public class MvScreenAdd
    {
        public int TenantId { get; set; }
        public required string Name { get; set; }
        public required string Address { get; set; }
        public required string Location { get; set; }
        public ScreenStatus Status { get; set; }
        public string Resolution { get; set; } = "1920x1080";
        public ScreenOrientation Orientation { get; set; }
        public List<string>? Tag { get; set; }
        public int? CreatedBy { get; set; }
    }

    public class MvScreenUpdate
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string Address { get; set; }
        public required string Location { get; set; }
        public ScreenStatus Status { get; set; }
        public string Resolution { get; set; } = "1920x1080";
        public ScreenOrientation Orientation { get; set; }
        public List<string>? Tag { get; set; }
        public int? UpdatedBy { get; set; }
    }

    public class MvScreenDelete
    {
        public int Id { get; set; }
        public int? DeletedBy { get; set; }
    }

    public class MvScreenFilter
    {
        public int TenantId { get; set; }
        public string? Search { get; set; }
        public int? Id { get; set; }
        public ScreenStatus? Status { get; set; }
        public ScreenOrientation? Orientation { get; set; }
    }

    public class MvScreenDropdown
    {
        public int Id { get; set; }
        public required string Name { get; set; }
    }

}

File: dooh-ad-manager/src/app/Application/INV/Screen/Models/screen.ts
import { ScreenStatus, ScreenOrientation, ScreenResolution } from "../../../../Shared/Models/enum.model";
import { MvScreenOperatingHour } from "./screen-operating-hour";

export class MvScreen {
    id : number;
    tenantId : number;
    name : string;
    address: string;
    location: string;
    resolution : ScreenResolution;
    status : ScreenStatus;
    orientation : ScreenOrientation;
    tag? : string[];
    createdAt? : Date;
    updatedAt? : Date;
    createdBy? : number;
    updatedBy? : number;
    isDeleted? : boolean;
    deletedAt? : Date;
    deletedBy? : number;
    selected?: boolean;
    operatingHours?: MvScreenOperatingHour[] | null; 
    

    constructor() {
    this.id = 0;
    this.tenantId = 1;
    this.name = '';
    this.address = '';
    this.location = '';
    this.resolution = ScreenResolution.R1920x1080;
    this.status = ScreenStatus.Active;
    this.orientation = ScreenOrientation.Landscape;
  }
}

export interface MvScreenAdd {
    tenantId : number;
    name: string;
    address: string;
    location: string;
    status: ScreenStatus;
    resolution : string;
    orientation : ScreenOrientation;
    tag?: string[];
    createdBy?: number;
}

export interface MvScreenUpdate {
    id: number;
    name: string;
    address : string;
    location: string;
    status: ScreenStatus;
    resolution: string;
    orientation: ScreenOrientation;
    tag?: string[];
    updatedBy? : number
}

export interface MvScreenDelete {
    id: number;
    deletedBy? : number;
}

export interface MvScreenFilter {
    tenantId : number;
    id? : number;
    search?: string;
    status?: ScreenStatus;
    orientation?: ScreenOrientation;
}

export interface MvScreenDropdown{
    id: number;
    name: string;
}